ui/packages/consul-ui/app/components/topology-metrics/index.js

```diff
@@ -108,14 +108,16 @@ export default class TopologyMetrics extends Component {
     // Calculate viewBox dimensions
     this.downView = document.getElementById('downstream-lines').getBoundingClientRect();
     const upstreamLines = document.getElementById('upstream-lines').getBoundingClientRect();
-    const upstreamColumn = document.getElementById('upstream-column').getBoundingClientRect();
-
-    this.upView = {
-      x: upstreamLines.x,
-      y: upstreamLines.y,
-      width: upstreamLines.width,
-      height: upstreamColumn.height,
-    };
+    if (document.getElementById('upstream-column')) {
+      const upstreamColumn = document.getElementById('upstream-column').getBoundingClientRect();
+
+      this.upView = {
+        x: upstreamLines.x,
+        y: upstreamLines.y,
+        width: upstreamLines.width,
+        height: upstreamColumn.height + 10,
+      };
+    }
```

Comment:
Are there any ways we can tighten this up a little, for example the code refers to the same DOM element twice, but looks up it up twice also.

Reply:
Where is it looked up twice? 🤔

Reply:
On lines 111 and 112:

```javascript
if (document.getElementById('upstream-column')) {
      const upstreamColumn = document.getElementById('upstream-column').getBoundingClientRect();
```